### PR TITLE
Improving Typescript

### PR DESCRIPTION
**Pull request by juanky201271**
Working on avoiding null, undefined, any... Build new types, etc...

### CODE REVIEW

**Comment by zancas** [comment]
Wow!   I love the 64 passing tests in CI!

https://github.com/zingolabs/zingo-mobile/actions/runs/3774222306/jobs/6416187025

File: __tests__/About.snapshot.tsx [hunk 1]
```diff
@@ -39,8 +45,8 @@ describe('Component About - test', () => {
   //snapshot test
   test('About - snapshot', () => {
     const state = {
-      navigation: null,
-      route: null,
+      navigation: {} as StackScreenProps<any>['navigation'],
```
Comment (zancas) [inline __tests__/About.snapshot.tsx]: Sweet!   So these are like...  empty instances of the type `StackScreenProps<any>['navigation']`?!

Comment (zancas) [reply __tests__/About.snapshot.tsx]: I have only a loose understanding of TypeScript, but these changes look great to me!

File: components/Transactions/components/TxSummaryLine.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ import moment from 'moment';
 type TxSummaryLineProps = {
   month: string;
   tx: Transaction;
-  setTxDetail: React.Dispatch<React.SetStateAction<Transaction | null>>;
+  setTxDetail: React.Dispatch<React.SetStateAction<Transaction>>;
```
Comment (zancas) [inline components/Transactions/components/TxSummaryLine.tsx:18]: I probably missed it, but does this mean there's also a change where many instances of `?` can be removed?

File: __tests__/About.snapshot.tsx (hunk 1 above)
Comment (juanky201271) [reply __tests__/About.snapshot.tsx]: Yes, you can initialize with an empty object, but declaring eyyyyy, here only accept this kind of value. Way better than `null`.

File: components/Transactions/components/TxSummaryLine.tsx (hunk 2 above)
Comment (juanky201271) [reply components/Transactions/components/TxSummaryLine.tsx:18]: Well, my first goal is/was to remove `null` as a valid value of everywhere, and in this case the value of this function could be `Transaction` or `null`. And now only accepts `Transaction`.

**Comment by juanky201271** [comment]
@zancas I think this is ready for merging... please.

**Review by zancas — APPROVED** [review]
Rubber Stamp

**Pull request merged**